src/object/srv_csum.c

```diff
@@ -541,8 +544,14 @@ ds_csum_add2iod(daos_iod_t *iod, struct daos_csummer *csummer,
 		daos_recx_t		*recx = &iod->iod_recxs[i];
 		struct dcs_csum_info	*info = &iod_csums->ic_data[i];
 
-		if (ctx.cc_rec_len > 0 && ci_is_valid(info))
+		if (ctx.cc_rec_len > 0 && ci_is_valid(info)) {
 			rc = cc_add_csums_for_recx(&ctx, recx, info);
+			if (rc != 0) {
+				D_ERROR("Failed to add csum for "
+	    					"recx"DF_RECX": %d\n",
```

Comment:
(style)  code indent should use tabs where possible
(style)  please, no space before tabs